wd_helper.au3

```diff
@@ -1102,6 +1102,99 @@ Func _WD_LoadWait($sSession, $iDelay = Default, $iTimeout = Default, $sElement =
+; Syntax ........: _WD_LocateElement($sSession, $sStrategy, $sSelector[, $bShadowRoot = Default])
+;                  $iErr = @error[, $iExt = @extended]]]])
+; Parameters ....: $sSession     - Session ID from _WD_CreateSession
+;                  $sStrategy    - Locator strategy. See defined constant $_WD_LOCATOR_* for allowed values
+;                  $sSelector    - $sSelector - Indicates how the WebDriver should traverse through the HTML DOM to locate the desired element(s).
+;                  $bShadowRoot  - [optional] Starting node is a shadow root? Default is False
+; Return values .: Success - array of matching frames (format like in _WD_FrameList)
+;                  Failure - "" (empty string) and sets @error to one of the following values:
+;                  - $_WD_ERROR_GeneralError
+;                  - $_WD_ERROR_Exception
+;                  - $_WD_ERROR_NoMatch
+; Author ........: mLipok
+; Modified ......:
+; Remarks .......: Returned location (path like 'null/2/0') can be used with _WD_FrameEnter before _WD_FindElement or _WD_WaitElement will be used.
+;                  In case when $_WD_ERROR_Exception is set returned location is valid, but was not able back to calling frame,
+;                  	or some frames have become inaccessible during processing
+; Related .......: _Wd_FrameList, _WD_FindElement
+; Link ..........:
+; Example .......: No
+; ===============================================================================================================================
+Func _WD_LocateElement($sSession, $sStrategy, $sSelector, $bShadowRoot = Default)
+	Local Const $sFuncName = "_WD_LocateElement"
+	Local Const $sParameters = 'Parameters:   Strategy=' & $sStrategy & '   Selector=' & $sSelector & '   ShadowRoot=' & $bShadowRoot
+	Local $iErr = $_WD_ERROR_Success
+	Local $sStartLocation = '', $sMessage = ''
+
+	; https://github.com/Danp2/au3WebDriver/pull/290#issuecomment-1100707095
+	; Prevent redundant logging from _Wd_FrameList and _WD_FrameEnter  and _WD_FindElement if not in Full debug mode
+	Local $_WD_DEBUG_Saved = $_WD_DEBUG ; save current DEBUG level
+	If $_WD_DEBUG <> $_WD_DEBUG_Full Then $_WD_DEBUG = $_WD_DEBUG_None
+
+	Local $aFrameList = _WD_FrameList($sSession, True)
+	$iErr = @error
+	If @error Then
+		$iErr = $_WD_ERROR_GeneralError
+		$sMessage = ' > Issue with getting list of frames'
+	Else
+		Local $iFrameCount = UBound($aFrameList, $UBOUND_ROWS)
+		For $i = 0 To $iFrameCount - 1
+			If $aFrameList[$i][$_WD_FRAMELIST_Relative] = '' Then $sStartLocation = $aFrameList[$i][$_WD_FRAMELIST_Absolute]
+		Next
+
+		For $i = $iFrameCount - 1 To 0 Step -1
+			_WD_FrameEnter($sSession, $aFrameList[$i][$_WD_FRAMELIST_Absolute])
+			If @error Then
+				$iErr = $_WD_ERROR_Exception
+				$sMessage = ' > Issue with entering frame=' & $aFrameList[$i][$_WD_FRAMELIST_Absolute]
+				ExitLoop
+			Else
+				_WD_FindElement($sSession, $sStrategy, $sSelector, Default, Default, $bShadowRoot)
```

Comment:
You should consider storing the result from `_WD_FindElement` in your array. This would allow the calling routine to switch to the target frame and then interact with the found element without having to call `_WD_FindElement` again.

Reply:
wondering if it will be ok when I decide to enlarge this:
```autoit
Global Enum _
		$_WD_FRAMELIST_Absolute = 0, _
		$_WD_FRAMELIST_Relative = 1, _
		$_WD_FRAMELIST_Attributes = 2, _
		$_WD_FRAMELIST_URL = 3, _
		$_WD_FRAMELIST_BodyID = 4, _
		$_WD_FRAMELIST_FrameVisibility = 5, _
		$_WD_FRAMELIST_MatchedElements = 6, _ ; array of matched element from _WD_LocateElement()
		$_WD_FRAMELIST__COUNTER
```

Reply:
When I suggested the above, I hadn't considered that multiple elements in the same frame could match the search criteria. That definitely complicated things. 🤔

Reply:
> I hadn't considered that multiple elements in the same frame could match the search criteria.

check it with:
https://www.tutorialspoint.com/html/html_frames.htm#

and: `li.nav-item[data-bs-original-title='Home Page'] a.nav-link[href='https://www.tutorialspoint.com/index.htm']`

I see 7 elements
![image](https://user-images.githubusercontent.com/11089482/223424752-df77d37b-6e44-4627-b79a-8301ac1c5c47.png)



> That definitely complicated things. 🤔

As so far I get such results:

![image](https://user-images.githubusercontent.com/11089482/223425729-20edac1d-d876-4684-ac25-4f8cc7276d9d.png)

Reply:
have some progress:
![image](https://user-images.githubusercontent.com/11089482/224577341-7a336d08-bfd1-41fd-8663-dd924e4ae851.png)

Reply:
IMO this part should be cosidered as Resolved